authorize/check_response.go

```diff
@@ -115,12 +115,13 @@ func invalidClientCertReason(reasons criteria.Reasons) bool {
 		reasons.Has(criteria.ReasonInvalidClientCertificate)
 }
 
-func (a *Authorize) okResponse(headers http.Header) *envoy_service_auth_v3.CheckResponse {
+func (a *Authorize) okResponse(headersToAdd http.Header, headersToRemove []string) *envoy_service_auth_v3.CheckResponse {
```

Comment:
naming nit:

```suggestion
func (a *Authorize) okResponse(headersToSet http.Header, headersToRemove []string) *envoy_service_auth_v3.CheckResponse {
```

(Based on https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/auth/v3/external_auth.proto#envoy-v3-api-field-service-auth-v3-okhttpresponse-headers it sounds like this would override an existing header?)